optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -213,6 +216,63 @@ private void applySolverProperties(RecorderContext recorderContext,
                 .collect(Collectors.toList());
     }
 
+    private void stopBuildIfClassesExistThatHavingPlanningAnnotationsButAreNotAPlanningEntity(SolverConfig solverConfig,
+            IndexView indexView) {
+        Collection<AnnotationInstance> optaplannerFieldAnnotations = new HashSet<>();
+
+        for (DotName annotationName : DotNames.PLANNING_ENTITY_FIELD_ANNOTATIONS) {
+            optaplannerFieldAnnotations.addAll(indexView.getAnnotations(annotationName));
+        }
+
+        BiFunction<AnnotationInstance, ClassInfo, String> errorMessage = (annotationInstance, classInfo) -> " with a @" +
+                annotationInstance.name().withoutPackagePrefix() + " annotation is in a class (" + classInfo.name().toString()
+                + ")" +
+                " that does not have a @PlanningEntity annotation. Maybe add a @PlanningEntity annotation to (" +
+                classInfo.name().toString() + ")?";
+        for (AnnotationInstance annotationInstance : optaplannerFieldAnnotations) {
```

Comment:
All the annotations enlisted in https://github.com/kiegroup/optaplanner/pull/1109/files#diff-016daa2e52bd778b35feb050f220a0024be4016ac03cf6d9b5c711e37caa2dc5R50 have `@Target({ METHOD, FIELD })`. Thus, the `CLASS`, `METHOD_PARAMETER` and `TYPE` cases never happen: the user will get a compilation error.

Reply:
Unless there is a bug in the Java compiler. What should the error message say? "Member annotation AT AnnotationName is placed on a TargetType, which is an invalid target for  AT AnnotationName. Maybe file a bug report for your compiler?"